### PR TITLE
Bump ci rocm6.2

### PR DESCRIPTION
**Pull request by TedThemistokleous**
Updating CI to ROCm 6.2 and other appropriate dockers

### CODE REVIEW

**Comment by TedThemistokleous** [comment]
Still waiting info for torch sha-1 used dont see it on repo.radeaon under the usual manylinux link. Will let this run over weekend in the meantime.

**Review by ahsan-ca — APPROVED** [review]

**Comment by ahsan-ca** [comment]
Looks good besides the CI failures.

**Comment by TedThemistokleous** [comment]
> Looks good besides the CI failures.

Yeah. something I'll have to deal with I think. @pfultz, should I be including some of your changes to the CPP checks from the 6.1.3 CI update PR in CMAkeLists.txt 
https://github.com/ROCm/AMDMIGraphX/pull/3335/commits/9291c014b94981b37e1a2241ff4d0630d0af9342

**Review by causten — APPROVED** [review]
You've got some more work to clear CI but the files I was expecting to see changes in are here.

**Comment by TedThemistokleous** [comment]
Rebased off #3335

File: src/include/migraphx/op/name.hpp [hunk 1]
```diff
@@ -37,7 +37,11 @@ struct op_name
 {
     std::string name() const
     {
+#if MIGRAPHX_ASAN
+        std::string name = get_type_name<Derived>();
+#else
         static const std::string& name = get_type_name<Derived>();
+#endif
```
Comment (pfultz2) [inline src/include/migraphx/op/name.hpp:44]: I dont think this should be here. I think there was a mistake with the merge.

File: src/include/migraphx/config.hpp [hunk 2]
```diff
@@ -56,3 +56,12 @@
...
 #endif // clang_tidy
+
+#if defined(__has_feature)
+#if __has_feature(address_sanitizer)
+#define MIGRAPHX_ASAN 1
+#endif
+#endif
+#ifndef MIGRAPHX_ASAN
+#define MIGRAPHX_ASAN 0
+#endif
```
Comment (pfultz2) [inline src/include/migraphx/config.hpp:67]: Yea this shouldn't be here either.

Comment (TedThemistokleous) [reply src/include/migraphx/config.hpp]: Removed

File: src/include/migraphx/op/name.hpp (hunk 1 above)
Comment (TedThemistokleous) [reply src/include/migraphx/op/name.hpp]: Removed

**Comment by TedThemistokleous** [comment]
Looks like somehow we're failing on an env variable due to something not setting hip_HIPCC_EXECUTABLE for this. Tryin to see if I modify the hip-clang docker if this will resolve the failures in tidy. I also updated the files the license check was complaining about.  I'll check back in the morning

**Comment by TedThemistokleous** [comment]
Odd, micimaster all passed but I dont see this reflected in PR. Looks like its just tidy/cpp check warnings/errors now?

**Comment by causten** [comment]
The pass is from 20 days ago.  I'll rekick. If you update the PR with a new commit it clears the Jenkins run information

**Review by CharlieL7 — APPROVED** [review]

**Comment by TedThemistokleous** [comment]
Got past the hipcc errors and the hip-config by modifying hip-clang.cmake usind sed to set the HIP_PLATFORM to "Amd"

Now hitting this

```
-- Generating embedding library 'migraphx_kernels'
-- Performing Test HAS_HIP_LAMBDA_HOST_DEVICE
-- Performing Test HAS_HIP_LAMBDA_HOST_DEVICE - Failed
-- Build with rocMLIR::rockCompiler 2.0.0
-- MIGraphX is using HIP Clang
CMake Error at cmake/TargetFlags.cmake:42 (get_target_property):
  get_target_property() called with non-existent target "hip::device".
Call Stack (most recent call first):
  cmake/TargetFlags.cmake:61 (get_target_property2)
  cmake/TargetFlags.cmake:100 (append_flags)
  src/targets/gpu/CMakeLists.txt:250 (target_flags)


-- Hip compiler flags: " "
-- Looking for miopenHiddenSetConvolutionFindMode in MIOpen
CMake Error at /data/build/CMakeFiles/CMakeTmp/CMakeLists.txt:20 (add_executable):
  Target "cmTC_cab74" links to target "hip::host" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at /usr/share/cmake-3.22/Modules/CheckLibraryExists.cmake:72 (try_compile):
  Failed to generate test project build system.
Call Stack (most recent call first):
  src/targets/gpu/CMakeLists.txt:291 (check_library_exists)
```

File: hip-clang.docker [hunk 3]
```diff
@@ -28,6 +28,7 @@ RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --allow-
     wget \
     rocm-device-libs \
     hip-dev \
+    hipcc \
```
Comment (pfultz2) [inline hip-clang.docker:31]: This should be added to the install_preqs.sh.

File: CMakeLists.txt [hunk 4]
```diff
@@ -196,6 +198,7 @@ rocm_enable_clang_tidy(
         clang-analyzer-alpha.cplusplus.MismatchedIterator
         clang-analyzer-alpha.cplusplus.MisusedMovedObject
         -bugprone-switch-missing-default-case
+        -bugprone-empty-catch
```
Comment (TedThemistokleous) [inline CMakeLists.txt:201]: created #3470

File: CMakeLists.txt [hunk 5]
```diff
@@ -172,8 +172,10 @@ rocm_enable_clang_tidy(
         -bugprone-easily-swappable-parameters
         -bugprone-implicit-widening-of-multiplication-result
         -bugprone-macro-parentheses
+	-bugprone-multi-level-implicit-pointer-conversion
         -bugprone-signed-char-misuse
         -bugprone-unchecked-optional-access
+	-bugprone-unused-local-non-trivial-variable
```
Comment (TedThemistokleous) [inline CMakeLists.txt:178]: Made #3471

**Pull request merged**